pkg/endpoint/endpoint.go

```diff
@@ -493,6 +495,19 @@ func createEndpoint(owner regeneration.Owner, policyGetter policyRepoGetter, nam
 	return ep
 }
 
+func (e *Endpoint) initDNSHistoryTrigger() {
+	// Note: This can only fail if the trigger func is nil.
```

Comment:
Incidentally, this is a pretty weird API, this feels like something we could catch at compile time rather than encoding runtime errors to catch it. Something to think about for future improvements.